core/server/proxy/src/main/java/alluxio/proxy/PathsRestServiceHandler.java

```diff
@@ -165,6 +170,27 @@ public Void call() throws Exception {
     }, ServerConfiguration.global());
   }
 
+  @GET
+  @Path(PATH_PARAM + DOWNLOAD_FILE)
+  @ReturnType("java.io.InputStream")
+  @Produces(MediaType.APPLICATION_OCTET_STREAM)
+  public Response downloadFile(@PathParam("path") final String path) {
+    AlluxioURI uri = new AlluxioURI(path);
+    Map<String, Object> headers = new HashMap<>();
+    headers.put("Content-Disposition", "attachment; filename=" + uri.getName());
+    return RestUtils.call(new RestUtils.RestCallable<InputStream>() {
+      @Override
+      public InputStream call() throws Exception {
+        FileInStream is =
+            mFileSystem.openFile(uri, OpenFilePOptions.getDefaultInstance());
+        if (is != null) {
+          return is;
+        }
+        throw new IllegalArgumentException("stream does not exist");
+      }
+    }, ServerConfiguration.global(), headers);
+  }
```

Comment:
Does the browser end up closing this stream?

Reply:
Ok, I added a log message in the streams "close" method and I see that it does log whenever I download a file.